auth0/resource_data.go

```diff
@@ -257,13 +257,25 @@ func Float64(d ResourceData, key string, conditions ...Condition) (f *float64) {
 // Bool accesses the value held by key and type asserts it to a pointer to a
 // bool.
 func Bool(d ResourceData, key string, conditions ...Condition) (b *bool) {
+	// We need to keep using this for bool values even if it's marked as deprecated.
+	// See https://github.com/hashicorp/terraform-plugin-sdk/issues/817.
 	v, ok := d.GetOkExists(key)
 	if ok && Any(conditions...).Eval(d, key) {
 		b = auth0.Bool(v.(bool))
 	}
 	return
 }
 
+// Flag accesses the value held by the bool flag key
+// and type asserts it as a pointer to a bool.
+func Flag(d ResourceData, key string, conditions ...Condition) (b *bool) {
+	v, ok := d.GetOk(key)
```

Comment:
The issue with the flags was that they were using GetOkExists to check if the bool value was set or not which actually defaults to false when the flag is not set forcing the entire payload to be sent as the flags are pointers to bool and omitempty will only omit nil values. 

Even tho GetOkExists is deprecated in the sdkv2 we need to keep using it for all the other bool fields we have on other structs. See https://github.com/hashicorp/terraform-plugin-sdk/issues/817 for more details.